File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.py
class Codec:
    d = {}
    a = 0
    def encode(self, longUrl: str) -> str:
        """Encodes a URL to a shortened URL.
        """
        self.d[self.a] = longUrl
        self.a+=1
        
        return str(self.a-1);
        

    def decode(self, shortUrl: str) -> str:
        """Decodes a shortened URL to its original URL.
        """
        return self.d[int(shortUrl)];
        

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))

File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.cpp
#include <string>
#include <unordered_map>
#include <map>
class Solution {
public:
    map<int, string>pq;
    // Encodes a URL to a shortened URL.
    int a;
    
    Solution(){
        a = 0;
    }
    string encode(string longUrl) {
        pair<int, string>p;
        p.first = a;
        p.second = longUrl;
        this->pq.insert(p);
        
        cout<<pq[a];
    
        return to_string(a++);
    }

    // Decodes a shortened URL to its original URL.
    string decode(string shortUrl) {
        return pq[stoi(shortUrl)];
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));